libraries/aws/jobs/src/aws_iot_jobs_serialize.c

```diff
@@ -1010,104 +1011,106 @@ static AwsIotJobsError_t _parseErrorDocument( const char * pErrorDocument,
                                    &pCode,
                                    &codeLength ) == false )
     {
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_JOBS_BAD_RESPONSE );
+        status = AWS_IOT_JOBS_BAD_RESPONSE;
     }
-
-    /* Match the JSON error code to a Jobs return value. Assume invalid status
-     * unless matched.*/
-    status = AWS_IOT_JOBS_BAD_RESPONSE;
```

Comment:
I believe deleting this line changes the behavior of the function, as now if the switch does not flow into the default, then `AWS_IOT_JOBS_STATUS_PENDING` can be returned instead of `AWS_IOT_JOBS_BAD_RESPONSE`.

You could save a few lines by initializing `status` to `AWS_IOT_JOBS_BAD_RESPONSE`, changing the `if` block to `== true`, and move the `switch` into the `if`

Reply:
You are totally right. It seems the AWS_IOT_JOBS_STATUS_PENDING will never return in original code.